### ISSUE TITLE
proto files compiled with go: inconsistent package names: tensorflow.grpc tensorflow

### ISSUE THREAD

**Issue by shallam66** [tensorflow/tensorflow#12445]
when I compile proto files with go, I got the error:
`protoc-gen-go: error:inconsistent package names: tensorflow.grpc tensorflow
--go_out: protoc-gen-go: Plugin failed with status code 1.`

my command:
`protoc -I=./tensorflow_serving  --go_out=plugins=grpc:. ./tensorflow/core/protobuf/*.proto`
tensorflow version: 1.3.0

I found that in the tensorflow/core/protobuf/master_service.proto and tensorflow/core/protobuf/worker_service.proto go package  declarations is 

> package tensorflow.grpc;

and other proto files in protobuf is

> package tensorflow.grpc;

go do not support different package names in one package。

**Comment by asimshankar** [comment]
Could you provide more detailed instructions on reproducing the problem? The exact sequence of commands to reproduce the environment and the problem would be helpful, as would some more context on what you're trying to do.

Having dots in the package names in the `.proto` file by itself is not a problem. According to [protocol buffer reference pages](https://developers.google.com/protocol-buffers/docs/reference/go-generated#package), the Go code generator handles dots in the package name by converting them to underscores.

The error you're seeing might be because you're trying to generate code for all `.proto` files in a single Go package. Depending on what you want to do, you might want to either generate code from only the `.proto` files that you need use multiple `protoc` invocations, each with a different subset of the `.proto` files.

**Comment by Ouwen** [comment]
@shallam66 I noticed that you are using the tensorflow serving protos with tensorflow protos. I had a similar issue when compiling protobufs when making a minimalist repo.

https://github.com/Ouwen/minflow_serve_proto

In this repo are the protos from tensorflow set in the original tree structure. I had to add
`option go_package` to all of `.proto` files which did not already have it.

Let me know if this fits what you are trying to do.
